subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,60 @@
+project('speexdsp', 'c',
+  version : '1.2.0',
+  default_options: ['warning_level=2',
+                    'c_std=c99',
+                    'buildtype=release'],
+  license : 'bsd-like')
+
+c_args = []
+cc = meson.get_compiler('c')
+host_cpu_family = host_machine.cpu_family()
+
+# Round up configuration data
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+cdata = configuration_data()
+
+# Add basic SIMD flags and config data
+if host_cpu_family.startswith('x86')
+  c_args += cc.get_supported_arguments(['-mfpmath=sse', '-msse2'])
+  cdata.set10('USE_SSE', true)
+  cdata.set10('USE_SSE2', true)
+
+elif host_cpu_family == 'arm'
+  c_args += cc.get_supported_arguments(['-mfpu=neon-vfpv4'])
+  cdata.set10('USE_NEON', true)
+
+elif host_cpu_family == 'aarch64'
+  c_args += cc.get_supported_arguments(['-mfpu=neon-fp-armv8'])
+  cdata.set10('USE_NEON', true)
+endif
+
+# Add any supported flags to the build
+foreach arg : c_args
+  add_project_arguments(arg, language: 'c')
+endforeach
+
+# C99 compilers (except for msvc) support variable-length arrays
+cdata.set('VAR_ARRAYS', cc.get_id() != 'msvc')
+
+# Configure EXPORT of symbols
+if cc.get_id() == 'msvc' and get_option('default_library') != 'static'
+  cdata.set('EXPORT', '')
+elif cc.has_argument('-fvisibility=hidden')
+  add_project_arguments('-fvisibility=hidden', language: 'c')
+  cdata.set('EXPORT', 'extern __attribute__ ((visibility ("default")))')
+else
+  cdata.set('EXPORT', 'extern')
+endif
```

Comment:
Instead of has_argument, please use [`cc.has_function_attribute()`](https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerhas_function_attribute)

Instead of using add_project_arguments, you can use `gnu_symbol_visibility: 'hidden'` on the actual library

Reply:
Thanks for the tips - both are now in place.

Reply:
From the method description:
> [This table](https://mesonbuild.com/Reference-tables.html#gcc-__attribute__) lists all of the supported attributes.

You just use the attribute name (`visibility`), not the name of some command-line flag. :)

Reply:
(eventually realized that.. and now switched back to github :slightly_smiling_face: and see your tip!)
Working good now.